controllers/component_build_controller_pac.go

```diff
@@ -407,6 +408,29 @@ func checkMandatoryFieldsNotEmpty(config map[string][]byte, mandatoryFields []st
 func (r *ComponentBuildReconciler) ensurePaCRepository(ctx context.Context, component *appstudiov1alpha1.Component, config map[string][]byte) error {
 	log := ctrllog.FromContext(ctx)
 
+	// Check multi-component git repository scenario
+	if component.Spec.Source.GitSource.Context != "" {
+		repository, err := r.findPaCRepositoryForComponent(ctx, component)
+		if err != nil {
+			return err
+		}
+		if repository != nil {
+			pacRepositoryOwnersNumber := len(repository.OwnerReferences)
+			if err := controllerutil.SetOwnerReference(component, repository, r.Scheme); err != nil {
+				log.Error(err, "failed to add owner reference to existing PaC repository", "PaCRepositoryName", repository.Name)
+				return err
+			}
+			if len(repository.OwnerReferences) > pacRepositoryOwnersNumber {
+				if err := r.Client.Update(ctx, repository); err != nil {
+					log.Error(err, "failed to update existing PaC repository with component owner reference", "PaCRepositoryName", repository.Name)
+					return err
+				}
+				log.Info("Added current component to owners of the PaC repository", "PaCRepositoryName", repository.Name, l.Action, l.ActionUpdate)
+			}
```

Comment:
else, should return error

and remove that return nil

Reply:
I believe, that the current logic is correct. We should not return error is everything is already up to date.

Reply:
I see, because that means that reference was already there